### ISSUE TITLE
HeaderMap issue

### ISSUE THREAD

**Issue by jdevoo** [adamwynne/twitter-api#69]
Hello,
I upgraded a previous project to 0.7.8 (and Clojure itself to 1.8) to find a problem with HeaderMap

```
#error {
 :cause IllegalName: compile__stub.clj_http.headers.clj-http.headers/HeaderMap
 :via
 [{:type clojure.lang.Compiler$CompilerException
   :message java.lang.NoClassDefFoundError: IllegalName: compile__stub.clj_http.
headers.clj-http.headers/HeaderMap, compiling:(clj_http/headers.clj:105:1)
   :at [clojure.lang.Compiler analyzeSeq Compiler.java 6875]}
  {:type java.lang.NoClassDefFoundError
   :message IllegalName: compile__stub.clj_http.headers.clj-http.headers/HeaderM
ap
   :at [java.lang.ClassLoader preDefineClass nil -1]}]
```

Reverting the [change ](https://github.com/adamwynne/twitter-api/commit/742ee61fabdbdeab75ce9daecb188c326d70c3e5)for ordering the header resolved it for me.

**Comment by bunzen** [comment]
I had the same problem. This solved it for me.

**Comment by infomofo** [comment]
Hi I'm running into the same issue reported here. Are there any suggested workarounds for this?

**Comment by castillobgr** [comment]
@infomofo's approach worked for me as well.

**Comment by chbrown** [comment]
I'm not sure why reverting that change, 742ee61, fixed it for you, since 742ee61 came [after v0.7.8](/adamwynne/twitter-api/compare/v0.7.8...742ee61). I expect that you pulled down master, reverted the commit, and then tested? In which case, pulling down master was what did the trick, not the reversion. :)

But I did run into that error with v0.7.8 as published to Clojars, and the `[clj-http "2.0.1"]` explicit (override) dependency fixed it for me.

In any case, bumping the `clj-oauth` version (0388e8e) from [1.5.1](https://clojars.org/clj-oauth/versions/1.5.1) to [1.5.5](https://clojars.org/clj-oauth/versions/1.5.5) appears to resolve this, published to Clojars minutes ago with [`[twitter-api "0.7.9"]`](https://clojars.org/twitter-api/versions/0.7.9).
